### PR TITLE
JBPM-7266 Configure jBPM unit tests to use connection pooling

### PR DESCRIPTION
**Pull request by rsynek**
Provides integration between jBPM engine, Narayana and Tomcat DBCP. Changes introduced by this PR have been already tested with various DBs and JEE containers.

Depends on https://github.com/kiegroup/drools/pull/1939

[DO NOT MERGE] - waiting for tomcat-dbcp stable release

### CODE REVIEW

**Comment by mswiderski** [comment]
> persistence.xml cannot directly set username and password due to [2]. Username and password must be set either on DataSourceXAConnectionFactory or the underlying XADataSource instance.

this is actually misconfiguration I'd say - since there is jta-data-source given in persistence.xml username and password in properties should anyway be ignored so I'd say let's remove it and rely only on data source instead.

**Comment by rsynek** [comment]
+1, will remove once we have something working

**Comment by mswiderski** [comment]
@rsynek how are we doing with this one? do we wait for JWS team at the moment?

**Comment by jhrcek** [comment]
@mswiderski just FYI @rsynek is on vacation till the ond of this week.

**Comment by rsynek** [comment]
@mswiderski yes, here is the issue I opened for this purpose https://issues.jboss.org/browse/JWS-1028

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
jenkins retest this

**Comment by zhfeng** [comment]
@rsynek  can you try my commit to fix the interposed transaction
https://github.com/zhfeng/drools/commit/a47255972e2dd6c83c721d50d67aaadce84a042d

**Comment by rsynek** [comment]
jenkins retest this

File: jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java [hunk 1]
```diff
@@ -135,6 +135,13 @@ private XADataSource createXaDataSource() {
...
             String url = driverProperties.getProperty("url", driverProperties.getProperty("URL"));
 
+            if (isH2()) {
+                final String username = driverProperties.getProperty("user");
+                final String password = driverProperties.getProperty("password");
+                xaDataSource.getClass().getMethod("setPassword", new Class[]{String.class}).invoke(xaDataSource, password);
+                xaDataSource.getClass().getMethod("setUser", new Class[]{String.class}).invoke(xaDataSource, username);
+            }
+
             if (!(className.startsWith("com.ibm.db2") || className.startsWith("com.sybase"))) {
```
Comment (triceo) [inline jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java]: Shouldn't this be rather `else if` then?

Comment (rsynek) [reply jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java]: There is already `else` branch on line https://github.com/rsynek/jbpm/blob/b5feebfc6fce286b5e66ef07b6fb80963bf0ec0b/jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java#L156 being currently executed for H2 too. I will think about making this part more readable - right now I wanted to see how many failures this fix eliminates.

Comment (triceo) [reply jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java]: Ok, that's fair.

Comment (rsynek) [reply jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java]: A bit of refactoring we talked about - in the target (not yet used) component https://github.com/web-servers/narayana-tomcat/pull/1

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
Failed probably due to jenkins being restarted.

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
The single failure is probably a random one; locally the test works.

**Comment by mswiderski** [comment]
@rsynek so what’s the next step for it?

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
jenkins retest this (testing with narayana-jta-5.8.1)

**Comment by rsynek** [comment]
Closing to merge https://github.com/kiegroup/drools/pull/1939 first. A new PR will replace this one soon.

**Pull request closed without merging**